modules/ding_adgangsplatformen/ding_adgangsplatformen.module

```diff
@@ -225,6 +243,41 @@ function ding_adgangsplatformen_logout($hooked = FALSE) {
   drupal_exit($logout_url);
 }
 
+
+/**
+ * Menu callback for iframes used by the single logout functionality.
+ *
+ * @see https://github.com/DBCDK/hejmdal/blob/master/docs/single-logout.md
+ */
+function ding_adgangsplatformen_logout_iframe() {
+  // Sites which iframe the single logout url can only access the content
+  // of the url. We cannot use default access control for this se we setup a
+  // response where the return code is mirrored in the content.
+  $status_code = 403;
+  $reason = 'Forbidden';
+  // Only log out SSO users. The user could in theory be logged in as normal.
+  // In this case Adgangsplatformen should not be allowed to logout the user.
+  if (ding_user_is_logged_in_with_sso()) {
+    $status_code = 200;
+    $reason = 'OK';
+
+    // We do not call user_logout_current_user() directly as this would cause
+    // ding_adgangsplatformen_logout() to redirect back to Adgangsplatformen.
+    session_destroy();
+  }
+
+  // Logout requests must never be cached. A cached response would cause
+  // Adgangsplatformen to think the user was logged out when it fact he was not.
+  drupal_page_is_cacheable(FALSE);
+  drupal_add_http_header('Status', "$status_code $reason");
+  drupal_add_http_header(
+    'X-Frame-Options',
+    'allow-from https://login.bib.dk'
```

Comment:
Maybe this should be put into configuration as an hidden variable as this endpoint have changed once before.

Reply:
@cableman: I see several options here.

1. Add a hidden variable
2. Use the hostnames of one of the urls from the OAuth provider (assuming that the iframe will be located on the same system.
3. A combination of 1 and 2 where the hidden variable falls back back on the hostname.

Reply:
@kasperg I was thinking something like https://github.com/ding2/ding2/blob/master/modules/ding_adgangsplatformen/includes/ding_adgangsplatformen.admin.inc#L43 so the solution was the same for these variables. That should not be changed but might need to be in the future...

Reply:
I have tried to implement this in da9824e.